image/defs.bzl

```diff
@@ -0,0 +1,90 @@
+load("@rules_swiftnav//image:transition.bzl", "multi_arch")
+load("@rules_pkg//:pkg.bzl", "pkg_tar")
+load("@rules_oci//oci:defs.bzl", "oci_image", "oci_image_index")
+
+def swift_image(name, bin = [], external_data = [], tars = [], **kwargs):
+    actual_tars = []
+
+    if bin:
+        tar_name = "{}_binary_tar".format(name)
+        pkg_tar(
+            name = tar_name,
+            srcs = bin,
+            package_dir = "/app/bin",
+            tags = ["manual"],
+        )
+        actual_tars.append(tar_name)
+
+    if external_data:
+        tar_name = "{}_external_data_tar".format(name)
+        pkg_tar(
+            name = tar_name,
+            srcs = external_data,
+            package_dir = "/app/external_data",
+            tags = ["manual"],
+        )
+        actual_tars.append(tar_name)
+
+    actual_tars += tars
+
+    oci_image(
+        name = name,
+        tars = actual_tars,
+        **kwargs
+    )
+
+def swift_image_index(name, image, platforms, **kwargs):
+    """
+    Creates a multi-arch image index
+
+    Args:
+        name: Name of the image target
+        image: Label of the image
+        platforms: List of platforms
+        **kwargs: Arguments passed down to oci_image_index
+    """
+    transition_name = "{}_transition".format(name)
+
+    multi_arch(
+        name = transition_name,
+        image = image,
+        platforms = platforms,
+    )
+
+    oci_image_index(
+        name = name,
+        images = [":{}".format(transition_name)],
+        **kwargs
+    )
+
+def swift_tag(name, value, out):
```

Comment:
suggestion on name: lets call this `image_tag` instead of `swift_tag`.

I think that's more expressive.

Also - should we name the parameter `value` or `tag`? I think the latter makes it more obvious, but maybe we're repeating the word `tag` to much. Up to you.

Last - Can we make `out` implicit? IIRC the convention in bazel is that if you don't explicitly need to specify the output artifact, then it should just be the name of the target (e.g. in `cc_binary` we don't need to specifically declare the `out` binary name).

Reply:
I get some warnings:
```
WARNING: /home/wiktor/Swift/orion/BUILD.bazel:1755:16: target 'stamp_tag' is both a rule and a file; please choose another name for the rule
WARNING: /home/wiktor/Swift/orion/BUILD.bazel:1760:10: target 'latest_master_tag' is both a rule and a file; please choose another name for the rule
WARNING: /home/wiktor/Swift/orion/BUILD.bazel:1765:10: target 'test_tag' is both a rule and a file; please choose another name for the rule
```


genrule's `outs` field is required, i guess i'll just postfix the filenames with .txt and it should be fine